deploy.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# This script automates the following steps:
+# - Tag a version on the main branch
+# - Update the corresponding docker-compose.yml on the prod server
+# - Restart the corresponding docker container on the prod server
+
+# Exit if a command failed
+set -e
+
+packagesUrl="https://github.com/mint-o-badges/badgr-server/pkgs/container/badgr-server"
+productionUrl="openbadges.education"
+productionUser="ubuntu"
+basePath="/home/ubuntu/docker"
+deploymentPath="$basePath/badgr-server"
+
+echo "Enter the commit checksum (e.g. 'f2ed4f8') you want to deploy (leave empty for the latest commit from the remote main)"
+read checksum
+echo "Enter the major part of the version you want to release (the x in vx.y.z)"
+read versionMajor
+echo "Enter the minor part of the version you want to release (the y in vx.y.z)"
+read versionMinor
+echo "Enter the patch part of the version you want to release (the z in vx.y.z)"
+read versionPatch
+echo "Enter a message you want to add to this tag"
+read message
+
+version="v$versionMajor.$versionMinor.$versionPatch"
+
+if [ -z "$checksum" ]; then
+    echo "Fetching latest commit..."
+    git fetch --all
+    checksum=$(git log -n 1 --pretty=format:"%H" origin/main)
+    echo "Tagging the latest commit ($checksum) with $version..."
+else
+    echo "Tagging commit $checksum with version $version..."
+fi
+git tag -a "$version" $checksum -m "$message"
+
+echo "Pushing tags to origin..."
+git push origin --tags
+
+echo -n "Waiting for build to finish..."
+while ! curl -s "$packagesUrl" | grep "$version" > /dev/null; do
```

Comment:
Do you think we should add something to handle a case where the curl command fails? Can this ```do while``` result in an infinite loop or am too scared maybe?

Reply:
What do you mean by fail? Which cases should we consider, in your opinion? E.g. if there's no internet connection, the commands before fail already. If github is down also (and then we'd have different problems). Another possibility is of course that the UI of github changes, thus indeed causing an infinite loop here. But I'm not sure how we'd check this. The alternative I could think of would be to use the github-cli program, but that needs to be installed first. Maybe I can check if I can find some kind of API feature from github for this though...

Reply:
Apparently there is an API endpoint for this, but in order to use it you need to have an access token with the `read:packages` scope. We thus have the following options:
1. Add a personal access token with this scope to this public repository
2. Handle the token as a secret, meaning that if someone wants to use the script, they need to add their own access token
3. Leave it as it currently is, not requiring a token

None of these options seems ideal. What do you think?

Reply:
I guess this is an unlikely edgecase, that we could still adapt to if it were to happen. I will approve of the script as it is